### PR TITLE
Code cleanup

### PR DESCRIPTION
**Pull request by jngrad**
Description of changes:
- fix various LGTM alerts and compiler warnings

### CODE REVIEW

File: src/core/unit_tests/Particle_test.cpp [hunk 1]
```diff
@@ -122,3 +122,72 @@ BOOST_AUTO_TEST_CASE(properties_serialization) {
     BOOST_CHECK_EQUAL(out.identity, prop.identity);
   }
 }
+
+void check_particle_force(ParticleForce const &out, ParticleForce const &ref) {
+  BOOST_CHECK_EQUAL(out.f[0], ref.f[0]);
+  BOOST_CHECK_EQUAL(out.f[1], ref.f[1]);
+  BOOST_CHECK_EQUAL(out.f[2], ref.f[2]);
```
Comment (KaiSzuttor) [inline src/core/unit_tests/Particle_test.cpp]:
```suggestion
  BOOST_TEST(out.f == ref.f, boost::test_tools::per_element() );
```

File: src/core/unit_tests/Particle_test.cpp [hunk 2]
```diff
@@ -122,3 +122,72 @@ BOOST_AUTO_TEST_CASE(properties_serialization) {
...
+#ifdef ROTATION
+  BOOST_CHECK_EQUAL(out.torque[0], ref.torque[0]);
+  BOOST_CHECK_EQUAL(out.torque[1], ref.torque[1]);
+  BOOST_CHECK_EQUAL(out.torque[2], ref.torque[2]);
```
Comment (KaiSzuttor) [inline src/core/unit_tests/Particle_test.cpp]:
```suggestion
```BOOST_TEST(out.torque == ref.torque, boost::test_tools::per_element() );```
```

**Review by KaiSzuttor — CHANGES_REQUESTED** [review]
Besides my comments, this can be merged

File: src/core/unit_tests/Particle_test.cpp (hunk 1 above)
Comment (jngrad) [reply src/core/unit_tests/Particle_test.cpp]: that's convenient, thanks!

**Review by KaiSzuttor — APPROVED** [review]

**Pull request merged**